# patch by Thomas E. Dickey <dickey@invisible-island.net>
# created  Sun Mar 29 23:16:16 UTC 2009
# ------------------------------------------------------------------------------
# Tekproc.c      |   25 ++++++++++++++-----------
# button.c       |    6 +++---
# charproc.c     |    4 ++--
# menu.c         |   10 +++++-----
# version.h      |    4 ++--
# xterm.h        |    4 ++--
# xterm.log.html |   14 +++++++++++++-
# 7 files changed, 41 insertions(+), 26 deletions(-)
# ------------------------------------------------------------------------------
Index: Tekproc.c
--- xterm-243+/Tekproc.c	2009-03-28 17:03:35.000000000 +0000
+++ Tekproc.c	2009-03-29 13:58:45.000000000 +0000
@@ -1,4 +1,4 @@
-/* $XTermId: Tekproc.c,v 1.167 2009/03/28 17:03:35 tom Exp $ */
+/* $XTermId: Tekproc.c,v 1.168 2009/03/29 13:58:45 tom Exp $ */
 
 /*
  * Warning, there be crufty dragons here.
@@ -596,7 +596,7 @@
 
 	case CASE_CHAR_SIZE:
 	    TRACE(("case: character size selector\n"));
-	    TekSetFontSize(tw, (int) (c & 03));
+	    TekSetFontSize(tw, False, (int) (c & 03));
 	    Tparsestate = curstate;
 	    break;
 
@@ -795,7 +795,7 @@
 	    tekRefreshList = tek;
 	    rptr = tek->data;
 	    rcnt = tek->count - 1;
-	    TekSetFontSize(tw, tek->fontsize);
+	    TekSetFontSize(tw, False, tek->fontsize);
 	    return (IChar) (*rptr++);
 	}
 	tekRefreshList = (TekLink *) 0;
@@ -938,7 +938,7 @@
 	tekscr->cur_X = 0;
 	tekscr->cur_Y = TEKHOME;
 	tekscr->cur = tekscr->page;
-	TekSetFontSize(tw, tekscr->cur.fontsize);
+	TekSetFontSize(tw, False, tekscr->cur.fontsize);
 	tekscr->margin = MARGIN1;
 	if (tekscr->TekGIN) {
 	    tekscr->TekGIN = NULL;
@@ -1649,7 +1649,7 @@
 }
 
 void
-TekSetFontSize(TekWidget tw, int newitem)
+TekSetFontSize(TekWidget tw, Bool fromMenu, int newitem)
 {
     if (tw != 0) {
 	TekScreen *tekscr = TekScreenOf(tw);
@@ -1666,7 +1666,8 @@
 	    set_tekfont_menu_item(oldsize, False);
 
 	    tekscr->cur.fontsize = newsize;
-	    tekscr->page.fontsize = newsize;
+	    if (fromMenu)
+		tekscr->page.fontsize = newsize;
 
 	    fid = tw->tek.Tfont[newsize]->fid;
 	    if (fid == DefaultGCID) {
@@ -1683,11 +1684,13 @@
 	    if (!Ttoggled)
 		TCursorToggle(tw, TOGGLE);
 
-	    /* we'll get an exposure event after changing fontsize, so we
-	     * have to clear the screen to avoid painting over the previous
-	     * text.
-	     */
-	    TekClear(tw);
+	    if (fromMenu) {
+		/* we'll get an exposure event after changing fontsize, so we
+		 * have to clear the screen to avoid painting over the previous
+		 * text.
+		 */
+		TekClear(tw);
+	    }
 	}
     }
 }
Index: button.c
--- xterm-243+/button.c	2009-03-27 00:00:56.000000000 +0000
+++ button.c	2009-03-29 17:04:40.000000000 +0000
@@ -1,4 +1,4 @@
-/* $XTermId: button.c,v 1.320 2009/03/27 00:00:56 tom Exp $ */
+/* $XTermId: button.c,v 1.321 2009/03/29 17:04:40 tom Exp $ */
 
 /*
  * Copyright 1999-2008,2009 by Thomas E. Dickey
@@ -1571,7 +1571,7 @@
 
     TRACE(("Getting %s (%ld)\n", name, (long int) type));
     for (cp = line; cp < line + len; cp++) {
-	TRACE(("[%d:%lu]", cp + 1 - line, len));
+	TRACE(("[%d:%lu]", (int) (cp + 1 - line), len));
 	if (isprint(*cp)) {
 	    TRACE(("%c\n", *cp));
 	} else {
@@ -3249,7 +3249,7 @@
     }
     *lp = '\0';			/* make sure we have end marked */
 
-    TRACE(("Salted TEXT:%d:%s\n", lp - line,
+    TRACE(("Salted TEXT:%d:%s\n", (int) (lp - line),
 	   visibleChars(PAIRED_CHARS(line, 0), (unsigned) (lp - line))));
 
     screen->selection_length = (unsigned long) (lp - line);
Index: charproc.c
--- xterm-243+/charproc.c	2009-03-29 00:23:12.000000000 +0000
+++ charproc.c	2009-03-29 22:14:41.000000000 +0000
@@ -1,4 +1,4 @@
-/* $XTermId: charproc.c,v 1.897 2009/03/29 00:23:12 tom Exp $ */
+/* $XTermId: charproc.c,v 1.898 2009/03/29 22:14:41 tom Exp $ */
 
 /*
 
@@ -398,7 +398,7 @@
     Bres(XtNallowFontOps, XtCAllowFontOps, screen.allowFontOp0, DEF_ALLOW_FONT),
     Bres(XtNallowTcapOps, XtCAllowTcapOps, screen.allowTcapOp0, DEF_ALLOW_TCAP),
     Bres(XtNallowTitleOps, XtCAllowTitleOps, screen.allowTitleOp0, DEF_ALLOW_TITLE),
-    Bres(XtNallowWindowOps, XtCAllowWindowOps, screen.allowWindowOp0, DEF_ALLOW_FONT),
+    Bres(XtNallowWindowOps, XtCAllowWindowOps, screen.allowWindowOp0, DEF_ALLOW_WINDOW),
     Bres(XtNaltIsNotMeta, XtCAltIsNotMeta, screen.alt_is_not_meta, False),
     Bres(XtNaltSendsEscape, XtCAltSendsEscape, screen.alt_sends_esc, False),
     Bres(XtNalwaysBoldMode, XtCAlwaysBoldMode, screen.always_bold_mode, False),
Index: menu.c
--- xterm-243+/menu.c	2009-03-28 17:27:57.000000000 +0000
+++ menu.c	2009-03-29 13:53:49.000000000 +0000
@@ -1,4 +1,4 @@
-/* $XTermId: menu.c,v 1.248 2009/03/28 17:27:57 tom Exp $ */
+/* $XTermId: menu.c,v 1.249 2009/03/29 13:53:49 tom Exp $ */
 
 /*
 
@@ -1570,7 +1570,7 @@
 		XtPointer closure GCC_UNUSED,
 		XtPointer data GCC_UNUSED)
 {
-    TekSetFontSize(getTekWidget(gw), tekMenu_tektextlarge);
+    TekSetFontSize(getTekWidget(gw), True, tekMenu_tektextlarge);
 }
 
 static void
@@ -1578,7 +1578,7 @@
 	    XtPointer closure GCC_UNUSED,
 	    XtPointer data GCC_UNUSED)
 {
-    TekSetFontSize(getTekWidget(gw), tekMenu_tektext2);
+    TekSetFontSize(getTekWidget(gw), True, tekMenu_tektext2);
 }
 
 static void
@@ -1586,7 +1586,7 @@
 	    XtPointer closure GCC_UNUSED,
 	    XtPointer data GCC_UNUSED)
 {
-    TekSetFontSize(getTekWidget(gw), tekMenu_tektext3);
+    TekSetFontSize(getTekWidget(gw), True, tekMenu_tektext3);
 }
 
 static void
@@ -1594,7 +1594,7 @@
 		XtPointer closure GCC_UNUSED,
 		XtPointer data GCC_UNUSED)
 {
-    TekSetFontSize(getTekWidget(gw), tekMenu_tektextsmall);
+    TekSetFontSize(getTekWidget(gw), True, tekMenu_tektextsmall);
 }
 
 static void
Index: version.h
--- xterm-243+/version.h	2009-02-28 15:56:11.000000000 +0000
+++ version.h	2009-03-29 13:49:44.000000000 +0000
@@ -1,4 +1,4 @@
-/* $XTermId: version.h,v 1.299 2009/02/28 15:56:11 tom Exp $ */
+/* $XTermId: version.h,v 1.300 2009/03/29 13:49:44 tom Exp $ */
 
 /*
  * These definitions are used to build the string that's printed in response to
@@ -6,7 +6,7 @@
  * version of X to which this version of xterm has been built.  The number in
  * parentheses is my patch number (Thomas E. Dickey).
  */
-#define XTERM_PATCH   243
+#define XTERM_PATCH   244
 
 #ifndef __vendorversion__
 #define __vendorversion__ "XTerm"
Index: xterm.h
--- xterm-243+/xterm.h	2009-03-28 14:56:54.000000000 +0000
+++ xterm.h	2009-03-29 13:54:54.000000000 +0000
@@ -1,4 +1,4 @@
-/* $XTermId: xterm.h,v 1.523 2009/03/28 14:56:54 tom Exp $ */
+/* $XTermId: xterm.h,v 1.524 2009/03/29 13:54:54 tom Exp $ */
 
 /************************************************************
 
@@ -681,7 +681,7 @@
 extern void TekRepaint (TekWidget /* xw */);
 extern void TekReverseVideo (TekWidget /* tw */);
 extern void TekRun (void);
-extern void TekSetFontSize (TekWidget /* tw */, int  /* newitem */);
+extern void TekSetFontSize (TekWidget /* tw */, Bool /* fromMenu */, int  /* newitem */);
 extern void TekSimulatePageButton (TekWidget /* tw */, Bool /* reset */);
 #endif
 
Index: xterm.log.html
--- xterm-243+/xterm.log.html	2009-03-29 00:29:55.000000000 +0000
+++ xterm.log.html	2009-03-29 22:16:27.000000000 +0000
@@ -20,7 +20,7 @@
  * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF   *
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.            *
  *****************************************************************************
-  $XTermId: xterm.log.html,v 1.744 2009/03/29 00:29:55 tom Exp $
+  $XTermId: xterm.log.html,v 1.746 2009/03/29 22:16:27 tom Exp $
   -->
 <HTML>
 <HEAD>
@@ -45,6 +45,7 @@
 is the latest version of this file.
 
 <UL>
+<LI><A HREF="#xterm_dev">Development</A>
 <LI><A HREF="#xterm_243">Patch #243 - 2009/3/28</A>
 <LI><A HREF="#xterm_242">Patch #242 - 2009/2/15</A>
 <LI><A HREF="#xterm_241">Patch #241 - 2009/1/26</A>
@@ -291,6 +292,17 @@
 <LI><A HREF="#xterm_01">Patch #1 - 1996/1/6</A>
 </UL>
 
+<H1><A NAME="xterm_dev">Development</A></H1>
+<ul>
+	<li>correct symbol used for default of <code>allowWindowOps</code>
+	    which was <code>DEF_ALLOW_FONT</code>
+	    rather than <code>DEF_ALLOW_WINDOW</code>
+	    (report by Matthieu Herrb).
+
+	<li>amend fix for tek4014 from <a href="#xterm_243">patch #243</a>
+	    to make it only apply to the Tek Options menu.
+</ul>
+
 <H1><A NAME="xterm_243">Patch #243 - 2009/3/28</A></H1>
 <ul>
 	<li>revert change to default for <code>allowTcapOps</code> (request by
